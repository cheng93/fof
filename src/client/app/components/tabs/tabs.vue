<template>
    <div>
        <nav class="mdc-tab-bar">
            <span v-for="(header, index) in headers"
                :key="index"
                v-on:click="selectIndex(index)"
                v-bind:class="{'mdc-tab--active': selectedIndex == index}"
                class="mdc-tab">
                {{header}}
            </span>
        </nav>
        <div v-for="(header, index) in headers"
            :key="index"
            v-if="selectedIndex == index">
            <slot :name="'tab-' + index"></slot>
        </div>
    </div>
</template>
<script>
export default {
    data() {
        return {
            selectedIndex: 0
        };
    },
    methods: {
        selectIndex: function(index) {
            this.selectedIndex = index;
        }
    },
    props: {
        headers: {
            type: Array,
            required: true
        }
    }
};
</script>
