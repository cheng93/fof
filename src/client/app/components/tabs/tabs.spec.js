import { mount } from '@vue/test-utils';
import tabs from './tabs.vue';

describe('Tabs', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(tabs, {
            propsData: {
                headers: ['A', 'B']
            },
            slots: {
                'tab-0': '<div>This is tab A</div>',
                'tab-1': '<div>This is tab B</div>'
            }
        });
    });

    it('matches snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('when clicked, should switch tab', () => {
        var otherTab = wrapper.find('.mdc-tab:nth-child(2)');
        otherTab.trigger('click');
        expect(wrapper.vm.selectedIndex).toBe(1);
    });

    it('when on other tab, should match snapshot', () => {
        wrapper.vm.selectedIndex = 1;
        expect(wrapper.html()).toMatchSnapshot();
    });
});
