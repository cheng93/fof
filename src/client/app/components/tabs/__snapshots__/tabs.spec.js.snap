// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs matches snapshot 1`] = `
<div>
  <nav class="mdc-tab-bar"><span class="mdc-tab mdc-tab--active">
            A
        </span><span class="mdc-tab">
            B
        </span></nav>
  <div>
    <div>This is tab A</div>
  </div>
  <!---->
</div>
`;

exports[`Tabs when on other tab, should match snapshot 1`] = `
<div>
  <nav class="mdc-tab-bar"><span class="mdc-tab">
            A
        </span><span class="mdc-tab mdc-tab--active">
            B
        </span></nav>
  <!---->
  <div>
    <div>This is tab B</div>
  </div>
</div>
`;
